app/models/targeting.rb

```diff
@@ -54,8 +54,9 @@ def static?
   end
 
   def self.build_query_from_hosts(ids)
-    hosts = Host.where(:id => ids).all.group_by(&:id)
-    hosts.map { |id, h| "name = #{h.first.name}" }.join(' or ')
+    return '' if ids.empty?
+    hosts = Host.where(:id => ids).all.group_by(&:name)
```

Comment:
What about going more sequel: `Host.where(:id => ids).distinct.pluck(:name)` to avoid unnecessary host objects creation?